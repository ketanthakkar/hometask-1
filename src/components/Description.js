import React from 'react';
import '../styleModules/Description.css';

const Description = ({desc}) => <p className = 'desc'> {desc} </p>;

export default Description;
