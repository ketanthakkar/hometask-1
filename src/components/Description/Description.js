import React from 'react';
import './Description.css';

const Description = ({desc}) => <p className = 'desc'> {desc} </p>;

export default Description;
