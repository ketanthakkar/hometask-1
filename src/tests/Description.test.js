import React from 'react';
import { shallow } from 'enzyme';
import Description from '../components/Description';


describe('<Description', () => {
    it('should lolololo', () => {
        const render = shallow(
            <Description desc = "description" />
        );
    
    expect(render.find('p').hasClass('desc')).toBe(true);
    expect(render).toMatchSnapshot();
    });

    it('should not modify props', () => {
      const desc = 'test';
      const render = shallow(
        <Description desc = {desc} />
      );
      expect(desc).toEqual('test');
      expect(render).toMatchSnapshot();
    })

});