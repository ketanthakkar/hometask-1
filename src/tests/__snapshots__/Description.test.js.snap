// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description should lolololo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description
    desc="description"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "description",
        " ",
      ],
      "className": "desc",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "description",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "description",
          " ",
        ],
        "className": "desc",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "description",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Description should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description
    desc="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "test",
        " ",
      ],
      "className": "desc",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "test",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "test",
          " ",
        ],
        "className": "desc",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "test",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
